backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/SqlRawQueryForSchema.java

```diff
@@ -20,26 +26,59 @@ public SqlRawQueryForSchema(SqlSchema schema) {
     this.schema = schema;
   }
 
-  public List<Row> executeSql(String sql) {
+  public List<Row> executeSql(String sql, Map<String, ?> parameters) {
     if (sql == null || sql.trim().equals("")) {
       return List.of();
     }
     List<Row> result = new ArrayList<>();
+
+    // convert parameters to sql bindings
+    List<Object> bindings = new ArrayList<>();
+    Pattern pattern = Pattern.compile("\\$\\{([^\\}]+)\\}"); // match ${xyz}
+    Matcher matcher = pattern.matcher(sql);
+    StringBuilder builder = new StringBuilder();
+    int i = 0;
+    while (matcher.find()) {
+      bindings.add(convertParameters(parameters, matcher.group(1)));
+      builder.append(sql.substring(i, matcher.start()));
+      builder.append("{" + (bindings.size() - 1) + "}");
+      i = matcher.end();
+    }
+    builder.append(sql.substring(i, sql.length()));
+    final String finalSql = builder.toString();
```

Comment:
sql injection ?

Reply:
I don't believe so. We replace our own ${} syntax with the {0} syntax of jooq. And then jooq takes care that only typed values are passed, not keywords or other stuff that you would need to inject sql.